script/scp-hosts.sh

```diff
@@ -47,11 +42,17 @@ do
   echo "scp dirs to $host/$installPath starting"
 	ssh -p $sshPort $host  "cd $installPath/; rm -rf bin/ conf/ lib/ script/ sql/ ui/"
 
+  for i in ${!workerNames[@]}; do
+    if [[ ${workerNames[$i]} == $host ]]; then
+      workerIndex=$i
+      break
+    fi
+  done
+
   for dsDir in bin master-server worker-server alert-server api-server ui
   do
-    # if worker in workersGroupMap
-    if [[ "${workersGroupMap[${host}]}" ]]; then
-      echo "export WORKER_GROUPS_0=${workersGroupMap[${host}]}" >> worker-server/bin/dolphinscheduler_env.sh
+    if [[ ! -z ${workerIndex+x} ]]; then
```

Comment:
Why do I need a default value of ```x``` here?

Reply:
It is a way to test whether workerIndex has been defined (The IP were included in workers or not ). If we don't need this I will remove it.

Reply:
Would it be better to use ```if [[ -n ${workerIndex} ]]```?

Reply:
Yes. I wrongly used two negative conditions (`! -z`). I will replace it.